infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/query/parsing/impl/InfinispanRemoteQueryResolverDelegate.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.infinispanremote.query.parsing.impl;
+
+import java.lang.invoke.MethodHandles;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.hibernate.hql.ast.common.JoinType;
+import org.hibernate.hql.ast.origin.hql.resolve.path.PathedPropertyReference;
+import org.hibernate.hql.ast.origin.hql.resolve.path.PathedPropertyReferenceSource;
+import org.hibernate.hql.ast.origin.hql.resolve.path.PropertyPath;
+import org.hibernate.hql.ast.spi.QueryResolverDelegate;
+import org.hibernate.ogm.datastore.infinispanremote.logging.impl.Log;
+import org.hibernate.ogm.datastore.infinispanremote.logging.impl.LoggerFactory;
+
+import org.antlr.runtime.tree.Tree;
+
+/**
+ * Query resolver delegate targeting Infinispan server queries. Very basic implementation atm.
```

Comment:
Let's remove the second sentence

Reply:
This class looks very similar to the other ones. Would it make sense to have a `BaseQueryResolverDelegate` and keep there all the common implementations?

Reply:
+1